### DISCUSSION TITLE
How data is handled after the gateway's transmission failed

### DISCUSSION THREAD

**Discussion by hufan-ola** [nats-io/nats-server#3546] [Q&A] [answered]
1. Does NATA Server will retry  or just drop this message?
2. How data is handled in the simplest PUB-SUB, the subscribe do not receive the data

**Comment by derekcollison** [comment, marked as answer]
In NATS core the messages will be lost if there is a loss in connectivity across a gateway.

If you do not want loss consider using JetStream which can survive network outages, self heals itself and provides at least once and *exactly* once (with caveats). NATS core provides at most once.
